### PR TITLE
ca: reuse consts instead of hardcode where applicable

### PR DESCRIPTION
**Pull request by kailun-qin**
Signed-off-by: Kailun Qin <kailun.qin@intel.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/bootstrap/certcontroller.go [hunk 1]
```diff
@@ -145,7 +146,7 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 		}
 		log.Infof("Generating istiod-signed cert for %v:\n %s", names, certChain)
 
-		signingKeyFile := path.Join(LocalCertDir.Get(), "ca-key.pem")
+		signingKeyFile := path.Join(LocalCertDir.Get(), ca.CAPrivateKeyID)
```
Comment (howardjohn) [inline pilot/pkg/bootstrap/certcontroller.go:149]: "ID" feels misleading here. Its not really an ID but a filename?

File: security/pkg/pki/ca/ca_test.go [hunk 2]
```diff
@@ -158,7 +157,7 @@ func TestCreateSelfSignedIstioCAWithSecret(t *testing.T) {
 	signingKeyPem := []byte(key1Pem)
 
 	client := fake.NewSimpleClientset()
-	initSecret := k8ssecret.BuildSecret("", CASecret, "default",
+	initSecret := BuildSecret("", CASecret, "default",
```
Comment (howardjohn) [inline security/pkg/pki/ca/ca_test.go:160]: The old code seemed right to me. Kubernetes specific abstractions (Secret) are decoupled from core code.

File: pilot/pkg/bootstrap/certcontroller.go (hunk 1 above)
Comment (kailun-qin) [reply pilot/pkg/bootstrap/certcontroller.go:149]: I agree, would making another const with the same value, and using correct semantics when applicable (i.e. file name) make sense to you? Something like: 
`const (`
    `CaCertID string "ca-cert.pem"`
    `CaCertFile`
`)`

Comment (kailun-qin) [reply pilot/pkg/bootstrap/certcontroller.go:149]: > When is ID every appropriate?

When building CASecret I imagine?

File: security/pkg/pki/ca/ca_test.go (hunk 2 above)
Comment (kailun-qin) [reply security/pkg/pki/ca/ca_test.go]: Not touching K8s Secret abstraction any more and make it self-contained. Would you please take another look @howardjohn? Thanks!

**Pull request merged**